### ISSUE TITLE
Callback instrument high CPU usage even without using it

### ISSUE THREAD

**Issue by jdanthinne** [AudioKit/AudioKitEX#5]
I've recently switched from the AppleSequencer to the Sequencer in order to build a metronome app (I was having the long-standing first measure bug with it).

Strangely, I was seeing the CPU usage going up a lot after the switch. After isolating various part of the code, it seems that using the CallbackInstrument was the cause of all this.

Is there something going on there? Here's a capture of Time Profiler, it seems that multiple threads are spawned.

<img width="496" alt="Capture d’écran 2022-01-17 à 13 31 51" src="https://user-images.githubusercontent.com/485107/149770152-d397e885-669c-4494-987b-82e5d975d0ac.png">

Here's the sample code I'm using if you want to test on your side.

Using AudioKit 5.3.0.

```swift
import AudioKit
import AudioKitEX
import UIKit

class ViewController: UIViewController {

    let engine = AudioEngine()
    let sequencer = Sequencer()
    var callbackTrack: SequencerTrack?

    override func viewDidLoad() {
        super.viewDidLoad()

        let callbacksInstrument = CallbackInstrument(midiCallback: { status, note, _ in
            print(status, note)
        })
        callbackTrack = sequencer.addTrack(for: callbacksInstrument)
        
        engine.output = callbacksInstrument
        try? engine.start()
    }

}
```

**Comment by aure** [comment]
Probably related to @Truba's change here https://github.com/AudioKit/AudioKit/pull/2632

**Comment by jdanthinne** [comment]
> Probably related to @Truba's change here [AudioKit/AudioKit#2632](https://github.com/AudioKit/AudioKit/pull/2632)

If the problem was running the NSTimer on the main thread, there was an option to run it on other thread (`[NSRunLoop currentRunLoop]`), and using a weak self for the memory leak.

**Comment by Truba** [comment]
Hi guys, 
@aure 
@jdanthinne 

I have to apologize, I haven't seen you mentioned here @aure .
My notifications don't seem to be always working.

The reason why I have even found this issue is that my code actually stopped working.
Code from this PR unfortunately doesn't work for any thread that is not main thread.
You also need to call `[[NSRunLoop currentRunLoop] run]` to start the timer.
I as goint to open another PR with that line, but I encountered another issue.
Issue with calling the `run` is that it blocks the current thread and if you are using a serial queue, it can block any added code to it.

Since I (and I assume most of iOS developers) mostly use GCD which handles the threading internally, I don't usually create NSThreads by hand. That's why I also like to avoid using RunLoops since they are really thread specific, and could cause issues when using with GCD which doesn't always guarantee the the same thread.

 I've also discovered the issue in my implementation:
`dispatch_source_set_timer(timer, DISPATCH_TIME_NOW, 0.01, 0);`
should be:
`dispatch_source_set_timer(timer, DISPATCH_TIME_NOW, 0.01 * NSEC_PER_SEC, 0);`
it was set to fire every 0.01 nanoseconds instead of 0.01 seconds.
That's why you received that CPU spike. 
Sry about that.

With all this said I would like open a PR with a revert to a revert 😄 
Hope that's ok with you guys?
